src/runtime/classobj.cpp

```diff
@@ -1927,6 +1927,18 @@ extern "C" PyObject* PyMethod_Self(PyObject* im) noexcept {
     return ((BoxedInstanceMethod*)im)->obj;
 }
 
+ extern "C" void PyMethod_SetSelf(PyObject* im, PyObject* instance) noexcept {
+     if (!PyMethod_Check(im)) {
+         PyErr_BadInternalCall();
+     }
+
+     Box* self = ((BoxedInstanceMethod*)im)->obj;
+     Py_XDECREF(self);
+     Py_INCREF(instance);
+     ((BoxedInstanceMethod*)im)->obj = instance;
+     Py_INCREF(im);
```

Comment:
I think the pattern is usually
```
Box* old = o->attr;
Py_INCREF(new_attr);
o->attr = new_attr;
Py_DECREF(old);
```
ie you want to decref the old value after replacing it, not before.  I also don't think we need the `Py_INCREF(im)` call here.